Add `no-plusplus` ESLint rule

ehmicky opened the pull request:
This adds the [`no-plusplus`](https://eslint.org/docs/rules/no-plusplus) ESLint rule.

erezrokah reviewed (changes requested):
What do you thing about enabling `allowForLoopAfterthoughts`?

ehmicky commented:
Good point! I just added this option.
This did feel a little silly to change all those `i++` to `i += 1` in `for` loops.
I have also changed it in the corresponding PR in Netlify Build.
